### PR TITLE
Fix links from a user's profile

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3354.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- on staging visit: /u/hatch
- click on juju-gui
- you should end up at /new/u/hatch/juju-gui/precise/14
- refresh the GUI
- the URL should remain the same and load correctly

**Comment by huwshimi** [comment]
[Fix store URLs from clicking charms in profiles.](https://trello.com/c/jAJZAqXk/85-fix-store-urls-from-clicking-charms-in-profiles)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8313/

File: jujugui/static/gui/src/app/state/test-state.js [hunk 1]
```diff
@@ -91,6 +91,10 @@ describe('State', () => {
     path: 'http://abc.com:123/new/store',
     state: { root: 'new', store: '' },
     error: null
+  }, {
+    path: 'http://abc.com:123/new/u/spinach/django/xenial/9',
```
Comment (hatched) [inline jujugui/static/gui/src/app/state/test-state.js]: Thanks

File: jujugui/static/gui/src/app/state/test-state.js [hunk 2]
```diff
@@ -1018,7 +1022,7 @@ describe('State', () => {
 
       const tests = ROOT_RESERVED.map(root => {
         return {
-          path: `http://abc.com:123/${root}/u/lukewh/cheese`,
```
Comment (hatched) [inline jujugui/static/gui/src/app/state/test-state.js]: This URL is also invalid on anything but `new`

Do we care to fail in that case? Maybe not...

File: jujugui/static/gui/src/app/state/state.js [hunk 3]
```diff
@@ -516,11 +516,17 @@ const State = class State {
     // If we have root paths in the URL then we can ignore everything else.
     if (state.root) {
       // If there is anything after this then it's an invalid URL unless it's the store.
-      if (parts.length > 1 && parts[1] !== 'store') {
+      if (parts.length > 1 && parts[1] !== 'store' &&
+        parts[1] !== PATH_DELIMETERS.get('user')) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/state/state.js]: Thanks, good find...

File: jujugui/static/gui/src/app/state/state.js [hunk 4]
```diff
@@ -516,11 +516,17 @@ const State = class State {
     // If we have root paths in the URL then we can ignore everything else.
     if (state.root) {
       // If there is anything after this then it's an invalid URL unless it's the store.
```
Comment (hatched) [inline jujugui/static/gui/src/app/state/state.js]: ... or the user delimiter

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8314/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**